test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -272,13 +272,16 @@ func kubeletCommand(kOp kubeletOpt, c clientset.Interface, pod *v1.Pod) {
 		framework.LogSSHResult(sshResult)
 	}
 	Expect(sshResult.Code).To(BeZero(), "Failed to [%s] kubelet:\n%#v", string(kOp), sshResult)
-	// On restart, waiting for node NotReady prevents a race condition where the node takes a few moments to leave the
-	// Ready state which in turn short circuits WaitForNodeToBeReady()
-	if kOp == kStop || kOp == kRestart {
+	if kOp == kStop {
 		if ok := framework.WaitForNodeToBeNotReady(c, pod.Spec.NodeName, NodeStateTimeout); !ok {
 			framework.Failf("Node %s failed to enter NotReady state", pod.Spec.NodeName)
 		}
 	}
+	if kOp == kRestart {
+		// On restart, Waiting for a minute before checking Node's Status to be Ready
+		framework.Logf("After restarting kubelet, waiting for a minute before checking Node's status to be Ready")
+		time.Sleep(NodeStateTimeout)
```

Comment:
instead of going directly to sleep, check node status first. Sleep only if it is not ready.

Reply:
@rootfs Added code to wait until Kubelet Pid is getting changed, and after the PID is changed, test waits until Node becomes ready.